src/main/webapp/intrigue-api/graphql.js

```diff
@@ -30,24 +45,55 @@ const createServerApollo = (...args) => {
 
   const cache = new InMemoryCache()
   return new ApolloClient({
-    link: new SchemaLink({
-      schema: executableSchema,
-      context: context(...args),
-    }),
+    link: ApolloLink.from([
+      serverErrorLink,
+      new SchemaLink({
+        schema: executableSchema,
+        context: context(...args),
+      }),
+    ]),
     ssrMode: true,
     cache,
   })
 }
 
+const clientErrorLink = onError(
+  ({ graphQLErrors, networkError, operation, forward }) => {
+    if (graphQLErrors) {
+      for (let err of graphQLErrors) {
+        if (err.extensions.code === 'UNAUTHENTICATED') {
+          //open modal and get credientials
+          const oldHeaders = operation.getContext().headers
```

Comment:
Once the credentials from the modal are passed, need to find the best way to permanently set the context to use that header.